libvirt/resource_libvirt_domain.go

```diff
@@ -239,6 +247,9 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	domainDef.OS.Type.Arch = d.Get("arch").(string)
+	domainDef.OS.Type.Arch = d.Get("machine").(string)
```

Comment:
This should be `domainDef.OS.Type.Machine`.

Reply:
Done I hope.

Sorry for the confusion, I don't often end up contributing to public projects via GitHub.